src/components/map/ThematicLayer.js

```diff
@@ -71,9 +71,10 @@ class ThematicLayer extends Layer {
         map.addLayer(this.layer);
 
         // Only fit map to layer bounds on first add
-        if (!editCounter) {
-            this.fitBounds();
-        }
+        // Disabled due to issues when no geometries was found on first try
+        // if (!editCounter) {
```

Comment:
This should probably check the "current" bounds and call `this.fitBounds()` if this layer's bounds fall "outside" the current bounds, right?

Maybe we can do that in a separate change, but this is technically a regression as it stands right?

Reply:
I'll have a second look :-)

Reply:
Related PR:https://github.com/dhis2/gis-api/pull/59

I've added a new getZoom method that can be used to check if the map is already zoomed. If not, we will always try to fit to layer bounds when the layer is updated.